File: libs/javascript/log.js
var process = require('process');
var fs = require('fs');

var log = function log(severity, component, message)
{
    var processID = process.pid;
    var threadID = 1;
    var ts = + new Date();

    if (!fs.existsSync(process.env.HOME+'/.logs'))
    {
	fs.mkdirSync(process.env.HOME+'/.logs');
    }
    try
    {
	message = message.toString();
    }
    catch (e)
    {
	severity = "ERROR";
	message = "Couldn't transform the object into a string."
    }
    fs.appendFile(process.env.HOME+'/.logs/logs', "["+ts+"]["+severity+"]["+processID+"]["+threadID+"]["+component+"]--"+message+"--\n", function (err) {
	if (err) throw err;
    });
}

log.trace = function(component, message)
{
    log("TRACE", component, message);
}

log.info = function(component, message)
{
    log("INFO", component, message);
}

log.debug = function(component, message)
{
    log("DEBUG", component, message);
}

log.warning = function(component, message)
{
    log("WARNING", component, message);
}

log.error = function(component, message)
{
    log("ERROR", component, message);
}

log.fatal = function(component, message)
{
    log("FATAL", component, message);
}

module.exports = log;

File: libs/python/Log.py
import os
import threading
import time

def     printlog(severity, component, *args):
    processID = os.getpid()
    threadID = threading.current_thread()
    timestamp = time.time() * 1000
    if not os.path.exists(os.getenv("HOME")+'/.logs'):
        try:
            os.makedirs(os.getenv("HOME")+'/.logs');
        except:
            pass
    try:
        message = "";
        for arg in args:
            message += str(arg)
    except:
        severity = "ERROR"
        message = "Message can not be converted as string."
    with open(os.getenv("HOME")+'/.logs/logs', 'ab') as f:
        f.write("[%d][%s][%s][%s][%s]--%s--\n"%(timestamp, severity, processID, threadID.ident, component, message))

def     trace(component, *args):
    printlog("TRACE", component, *args)

def     debug(component, *args):
    printlog("DEBUG", component, *args)

def     info(component, *args):
    printlog("INFO", component, *args)

def     warning(component, *args):
    printlog("WARNING", component, *args)

def     error(component, *args):
    printlog("ERROR", component, *args)

def     fatal(component, *args):
    printlog("FATAL", component, *args)

